File: tests/uasm/sample/nested_if_merge/NestedIfMergeTest.cs
using UdonSharp;
using UnityEngine;

[UdonBehaviourSyncMode(BehaviourSyncMode.None)]
public class NestedIfMergeTest : UdonSharpBehaviour
{
    public void Start()
    {
        int score = 9;
        int lives = 1;

        string result = "";
        if (score > 10)
        {
            result = "clear";
        }
        else if (lives > 0)
        {
            result = "retry";
        }
        else
        {
            result = "fail";
        }

        Debug.Log(result);
    }
}

File: tests/uasm/sample/nested_if_merge/NestedIfMergeTest.ts
import { UdonBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonDecorators";
import { UdonSharpBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonSharpBehaviour";
import { Debug } from "@ootr/udon-assembly-ts/stubs/UnityTypes";

@UdonBehaviour()
export class NestedIfMergeTest extends UdonSharpBehaviour {
  Start(): void {
    const score = 9;
    const lives = 1;

    let result = "";
    if (score > 10) {
      result = "clear";
    } else if (lives > 0) {
      result = "retry";
    } else {
      result = "fail";
    }

    Debug.Log(result);
  }
}
